### DISCUSSION TITLE
SAML authentication msg signature validation bypass vulnerability

### DISCUSSION THREAD

**Discussion by srd90** [requarks/wiki#5763] [Error / Bug Report] [answered]
See: https://github.com/node-saml/passport-saml/security/advisories/GHSA-m974-647v-whv7

Workaround (according to vulnerability report): Disable SAML authentication.

wikijs is using at the moment vulnerable version:
https://github.com/requarks/wiki/blob/98e199a0095a2b520133844051bfb3bf150efe5e/package.json#L151

@NGPixel

**Comment by NGPixel** [comment, marked as answer]
Version updated in 504f1725a2d497bf5a62416f0ae2cc890359c500, part of 2.5.290.

Please use the methods detailed in the [security policy](https://github.com/requarks/wiki/security/policy) next time, to get this addressed in a speedy manner.

**Reply by srd90** [reply]
@NGPixel You do not follow `security@requarks.io`. I did send note to that email address. Although this time I sent note first into these discussions.

About previous similar case ( #5180 ):
1. `January, 2021`: sent note to aforementioned email address
2. `Apr 12, 2022`:  (**over a year later**) created: #5180 
3. `May 1, 2022`: sent another note: https://github.com/requarks/wiki/discussions/5180#discussioncomment-2669477

About this case ( #5763 ):
1. `Oct 19, 2022`: When I created this particular discussion the [CVE](https://github.com/node-saml/passport-saml/security/advisories/GHSA-m974-647v-whv7) had already existed several days (i.e. wikijs' situation had been visible several days via various tools)
2. `Oct 24, 2022`: Sent a note to `security@requarks.io`
3. `Oct 30, 2022`: You updated vulnerable dependency

Here is the content of `Oct 24, 2022` email:
>Hi,
>
>https://github.com/requarks/wiki/discussions/5763
>
>at the time of writing aforementioned discussion message vulnerability had already been visible/discoverable several days via various scanning tools (incl. but not limited to snyk, npm audit, owasp depedency-check etc.)
>
>Furthermore:
>
>based on "npm audit" wiki.js has:
>102 vulnerabilities (2 low, 58 moderate, 38 high, 4 critical)
> 
>and "npm audit --production":
>39 vulnerabilities (2 low, 16 moderate, 18 high, 3 critical)
>
>some of those seems to be related to remote code execution.

It seems that this project updates pinned dependencies only in case someone reports need to update instead of proactively reacting to e.g. output of npm install which gives red flags about dependencies which have public CVEs. 

Note: There is 5 month gap in dependency updates ( see `May 2022 - Oct 30, 2022`: https://github.com/requarks/wiki/commits/main/package.json ).
May 2022 dependency updates were triggered by #5180 and Oct 2022 updates were triggered by #5763

**Reply by NGPixel** [reply]
@srd90 My bad, I just rechecked and I did receive your email, but it got lost in the bulk.

As for the pinned dependencies, this is a bit tricky as updating them to the latest version would result in a broken app and require some major refactoring in many cases. This situation will be improved in 3.0.
